ledger/src/blockstore.rs

```diff
@@ -2042,9 +2069,22 @@ impl Blockstore {
                     .last()
                     .map(|entry| entry.hash)
                     .unwrap_or_else(|| panic!("Rooted slot {slot:?} must have blockhash"));
+                let mut starting_transaction_index = 0;
+                let mut entries = vec![];
                 let slot_transaction_iterator = slot_entries
                     .into_iter()
-                    .flat_map(|entry| entry.transactions)
+                    .flat_map(|entry| {
+                        if populate_entries {
+                            entries.push(solana_transaction_status::EntrySummary {
+                                num_hashes: entry.num_hashes,
+                                hash: entry.hash.to_string(),
```

Comment:
Is stringification of the transactions (signature & pubkeys presumably) done by the caller / at a higher level? Ideally, I think all of the stringification would occur at the same level. Given that we already have a mix, I don't feel too strongly about this one, but maybe a nice cleanup task after would be to shift all stringification to the same level. Can we even do that at this point, or would be break some types that we really shouldn't break?

Reply:
I looked at the existing BigTable protobuf definitions more thoroughly. Signatures are stored as `bytes` (and stringified before being returned by RPC). Pubkeys are actually stored a mix of `string` and `bytes`, sigh. And the only existing hashes in the proto definitions are blockhashes, which are stored as `strings`.
Since there's no clear pattern to follow, I'm now inclined to go with the type that will, on average, use less space, which is `bytes` and leave stringification to any potential readers.

> Can we even do that at this point, or would be break some types that we really shouldn't break?

I think it is hypothetically possible to rewrite all the protobuf `string` types to `bytes`, but it would be a pita to support multiple readers for the old and new versions.

Reply:
> Since there's no clear pattern to follow, I'm now inclined to go with the type that will, on average, use less space, which is bytes and leave stringification to any potential readers.

Works for me

> I think it is hypothetically possible to rewrite all the protobuf string types to bytes, but it would be a pita to support multiple readers for the old and new versions.

Yeah, not a huge deal / not worth the effort to me either. I'm good with this one being marked as resolved

Reply:
Done!